includes/db.inc

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/**
+ * @file
+ * Database helper functions.
+ */
+
+/**
+ * Get the proxies for a given PID.
+ */
+function islandora_owner_permissions_get_proxies_for_pid($pid) {
+  $pid_id = islandora_owner_permissions_get_pid_id($pid);
+  return db_select('islandora_owner_permissions_object_proxies', 'proxies')
+    ->fields('proxies', array('uid'))
+    ->condition('object', $pid_id)
+    ->execute()
+    ->fetchAllAssoc('uid', PDO::FETCH_ASSOC);
+}
+
+/**
+ * Add a proxy to a given PID.
+ */
+function islandora_owner_permissions_add_proxy_to_pid($pid, $user_name) {
+  $user = user_load_by_name($user_name);
+  $pid_id = islandora_owner_permissions_get_pid_id($pid);
+  return db_insert('islandora_owner_permissions_object_proxies')
+    ->fields(array('object' => $pid_id, 'uid' => $user->uid))
+    ->execute();
+}
+
+/**
+ * Remove a proxy from a given PID.
+ */
+function islandora_owner_permissions_remove_proxy_from_pid($pid, $uid) {
+  $pid_id = islandora_owner_permissions_get_pid_id($pid);
+  db_delete('islandora_owner_permissions_object_proxies')
+    ->condition('uid', $uid)
+    ->condition('object', $pid_id)
+    ->execute();
+}
+
+/**
+ * Remove a user from the db.
+ */
+function islandora_owner_permissions_remove_user_from_db($uid) {
+  db_delete('islandora_owner_permissions_object_proxies')
+    ->condition('uid', $uid)
+    ->execute();
+}
+
+/**
+ * Remove an object from the db.
+ */
+function islandora_owner_permissions_remove_object_from_db($pid) {
+  $pid_id = islandora_owner_permissions_get_pid_id($pid);
+  db_delete('islandora_owner_permissions_objects')
+    ->condition('id', $pid_id)
+    ->execute();
+  db_delete('islandora_owner_permissions_object_proxies')
+    ->condition('object', $pid_id)
+    ->execute();
+}
+
+/**
+ * Gets the database id related to a PID.
+ *
+ * Sets one if it doesn't exist.
+ */
+function islandora_owner_permissions_get_pid_id($pid) {
+  $id = islandora_owner_permissions_pid_id($pid);
+  // If no matches, create one.
+  if (!$id) {
+    return islandora_owner_permissions_pid_to_db($pid);
+  }
+  return $id;
+}
+
+/**
+ * Creates/gets the database id related to a PID.
+ *
+ * @throws Exception if the object cannot be loaded.
+ */
+function islandora_owner_permissions_pid_to_db($pid) {
+  $object = islandora_object_load($pid);
```

Comment:
Not too terribly concerned, but do we particularly care about this?... Could hypothetically be useful to call before the object is loadable, say, from an ingest step.

Reply:
I would like to avoid it.

Reply:
Okay.